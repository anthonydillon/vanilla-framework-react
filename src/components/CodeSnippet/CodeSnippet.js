import React from 'react';
import PropTypes from 'prop-types';
import './CodeSnippet.css';

class CodeSnippet extends React.Component {
  render() {
    return (
      <div className="p-code-snippet">
        <input className="p-code-snippet__input" value={this.props.value} readOnly="readonly" />
        <button className="p-code-snippet__action">Copy to clipboard</button>
      </div>
    );
  }
}

CodeSnippet.propTypes = {
  value: PropTypes.string.isRequired,
};

export default CodeSnippet;
