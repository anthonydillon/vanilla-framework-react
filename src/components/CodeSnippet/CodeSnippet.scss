@import 'vanilla-framework/scss/base_code';
@import 'vanilla-framework/scss/patterns_code-snippet';
@include vf-p-code-snippet;