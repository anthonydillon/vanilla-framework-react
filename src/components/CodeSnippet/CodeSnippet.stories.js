import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import CodeSnippet from './CodeSnippet';

storiesOf('Code Snippet', module)
  .add('Default',
    withInfo('Code snippet should be used when presenting the user with a small snippet of code that they will likely want to copy and paste.')(() =>
      <CodeSnippet value="sudo apt-get update" />,
    ),
  );
