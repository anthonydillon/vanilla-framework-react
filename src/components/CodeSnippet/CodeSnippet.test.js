import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import CodeSnippet from './CodeSnippet';

describe('CodeSnippet component', () => {
  it('should compare with a snapshot', () => {
    const codesnippet = ReactTestRenderer.create(<CodeSnippet value="sudo apt-get update" />);
    const json = codesnippet.toJSON();
    expect(json).toMatchSnapshot();
  });
});

