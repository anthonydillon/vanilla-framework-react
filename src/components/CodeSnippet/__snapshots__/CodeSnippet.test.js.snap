// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeSnippet component should compare with a snapshot 1`] = `
<div
  className="p-code-snippet"
>
  <input
    className="p-code-snippet__input"
    onClick={[Function]}
    readOnly="readonly"
    value="sudo apt-get update"
  />
  <button
    className="p-code-snippet__action"
    onClick={[Function]}
  >
    Copy to clipboard
  </button>
</div>
`;
